src/views/blocks/CarouselBlock.php

```diff
@@ -1,32 +1,48 @@
 <?php
 /* @var $this \luya\cms\base\PhpBlockView */
-
-if (!$this->isPrevEqual) {
-    $this->appView->registerJs("$('.carousel').carousel()");
-}
+$this->appView->registerJs("$('.carousel').carousel(".$this->extraValue('conf').")");
+$id = $this->extraValue('id');
+$images = $this->extraValue('images');
+$first = true;
+$indicators = '';
+$counter = 0;
 ?>
-<?php if (!$this->getIsPrevEqual()): ?>
-<div id="<?= $this->extraValue('id'); ?>" class="carousel slide" data-ride="carousel">
-	<div class="carousel-inner">
-<?php endif; ?>
-	<?php if ($this->extraValue('image')): ?>
-    <div class="carousel-item <?php if (!$this->getIsPrevEqual()): ?>active<?php endif; ?>">
-      	<img class="d-block w-100" src="<?= $this->extraValue('image')->source; ?>" alt="<?= $this->varValue('title'); ?>">
-      	<div class="carousel-caption d-none d-md-block">
-    	<h5><?= $this->varValue('title')?></h5>
-    	<p><?= $this->varValue('caption'); ?></p>
-  		</div>
-    </div>
+
+<div id="<?= $id ?>" class="carousel slide<?= $this->cfgValue('crossfade', null , ' carousel-fade'); ?><?= $this->cfgValue('row', null, ' row') ?>" data-ride="carousel">
+    <div class="carousel-inner">
+    <?php if ($images):
+        foreach ($images as $image):
```

Comment:
maybe move $counter++ to `foreach ($images as $image): $counter++;` makes it more easy to read.